packager/src/main/java/io/cdap/hub/publisher/GCSPublisher.java

```diff
@@ -80,15 +80,24 @@ public void publish(Hub hub) throws Exception {
     for (CategoryMeta categoryMeta : hub.getCategories()) {
       publishCategory(categoryMeta);
     }
+
+    StringBuilder keyBuilder = new StringBuilder();
+    if (prefix != null && !prefix.equals("")) {
```

Comment:
I would also recommend using a `Path` rather than a String, for concatenation. Be careful though: it can get complex with GCS URIs. Ingestion created this lib for those: https://github.com/LiveRamp/ingestion_lib/blob/925227bf6b41488e4dc78b36210fb36804f80ab4/src/java/com/liveramp/ingestion_lib/utils/PathCreator.java#L12

Reply:
Agreed with Baptiste -- my only comment is not to add a dependency on an LR library so we can still push `GCSPublisher` into OSS when done.